docs/standards/generic-standards/lsp1-universal-receiver.md

```diff
@@ -35,26 +45,50 @@ This standard defines a single function named `universalReceiver(...)` that coul
 - bytes32 `typeId`: Hash or Hook of a specific standard.
 - bytes `data`: Any arbitrary data.
 
-The `universalReceiver(...)` function emits an event with the data passed to it and some additional data. The function can then implement custom logic to make the contract behave differently based on the data received. For instance, the universalReceiver(...) function offers the following possibilities:
+> Receiving contracts should consider the `typeId` parameter to **decode the data correctly**.
+
+The `universalReceiver(...)` function **emits an event with the data passed to it and some additional data**. 
+
+![universalReceiver function emits an event](/img/standards/lsp1/universal-receiver-event.jpeg)
+
+For example, **to mark the receiving of a specific token**, during a token transfer, the token contract can call the `universalReceiver(..)` function of the token recipient with:
+
+- bytes32 `typeId`: **Hash**('ERCXXXXTokenReceived')
+- bytes `data`: **packedData**(amount of token sent, the sender address, the block timestamp)
+
+In this way, instead of **listening to all the events of the token contrats on the network**, and checking which one of these transfers is relative to the recipient, users can listen to the **[UniversalReceiver](../smart-contracts/lsp0-erc725-account.md#universalreceiver-1)** event on the contract implementing the `universalReceiver(..)` and know the token transfer details.  
+
+
+Also, the function can then implement **custom logic** to make the contract behave differently based on the data received. For instance, the `universalReceiver(...)` function, if customized, can do the following:
 
 - Reverting on calls to completely disallow the smart contract from receiving assets, information, etc. :x:
 - Registering the received assets inside the contract storage (see [LSP5 - Received Assets](../universal-profile/lsp5-received-assets.md)). :heavy_plus_sign:
```

Comment:
I suggest being as creative as possible here 🎨 🙂 , with multiple ideas to really explain **how extensible LSP1 is:**

- Disallow receiving specific tokens from specific token contract addresses, for instance (_e.g: spam tokens)._
- Forward and store specific tokens in a [LSP9 Vault]("link to LSP9 Standard").
- Stake-specific tokens in a staking contract (e.g: governance tokens).
- Forward specific tokens in a contract behind a protocol or dApp (e.g: liquidity or lending pool to earn interest).
- Depending on the `typeId`, save a percentage % of tokens received (native tokens or not), by placing them in an LSP9 Vault for instance.

Reply:
Good ideas, will not specify LSP9Vault as it could be overcomplicated for new users reading the standard, will keep it generic and write vaults